README.md

```diff
@@ -1,2 +1,105 @@
 # elixir-base58-encode
 https://hex.pm/packages/b58
+
+
+This module provides the `Base58Encode.encode/1` function which takes an **Elixir binary** and returns its base58 String representation.
+
+See the section [What is an Elixir binary?](#What-is-an-elixir-binary) for more information about the binary type in Elixir.
+
+## How to use the module
+
+1. Add the package to you dependencies
+
+  ```
+  defp deps do
+    [b58: "~> 0.1.0"]
+  end
+  ```
+
+  and run `mix deps.get`
+
+2. Call the `encode` function with a binary as parameter:
+
+ ```
+ Base58Encode.encode("foo")
+ "bQbp"
+ ```
+
+ if the parameter is not an Elixir binary the function will return `:error`. In
+ this example 42 is an Integer and not a binary (`is_binary(42)` will return `false`)
+
+ ```
+ Base58Encode.encode(42)
+ :error
+ ```
+
+ See `example/Example.exs` file for a simple example on how to use the module.
+ You can also run this example with `mix run example/Example.exs`
+
+## What is base58?
+
+`base58` provides a way to represent an **integer to a string** where the characters of the string are selected from the following list:  
+
+`123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz`
+
+The idea is to map each character of the set to a value: 0 to 1, 2 to 2 ... 57 to z.
```

Comment:
Is this correct...
```
0 to 1, 2 to 2 ... 57 to z
```

If it's `<<0>> to "1"` wouldn't `<<1>> to "2"` make sense and `<<2>> to "3"`. (Also think you should include the `""` and `<<>>` to clearly show which is the binary and which is the result)

`The idea is to map each character` I think that we should change the ` map` to `match` because I think it is more clear (But I am not great at writing clear readmes so a second opinion would be good)

Reply:
yes you are right for 1 match 2 in base58.
However we don't match directly elixir binaries here. When we convert an integer to base58 the returned result is a sequence of numbers (between 0 and 57). Then each number of this sequence is match to the base58 character representation.
I'll make this a bit more obvious in the Readme with `0 -> "1", 1 -> "2", ..., 57 -> "z"`